### ISSUE TITLE
Configuration for rule "one-var" is invalid:

### ISSUE THREAD

**Issue by stevenvachon** [eslint/eslint#10200]
This is my config:
```json
"one-var": [2, {
    "const": "never",
    "initialized": "never",
    "uninitialized": "consecutive"
}],
```

… which is to support and enforce the following:
```js
const a = 'value';
const b = 'value';
let c = 'value';
let d = 'value';
something(() => {
    c = 'new value';
    d = 'new value';
});

let e, f, g, h;
something(() => {
    e = 'new value';
    f = 'new value';
    g = 'new value';
    h = 'new value';
});
```
… but throws:
```shell
Error: /project/.eslintrc:
	Configuration for rule "one-var" is invalid:
	Value {"const":"never","initialized":"never","uninitialized":"consecutive"} should be equal to one of the allowed values.
	Value {"const":"never","initialized":"never","uninitialized":"consecutive"} should NOT have additional properties.
	Value {"const":"never","initialized":"never","uninitialized":"consecutive"} should NOT have additional properties.
	Value {"const":"never","initialized":"never","uninitialized":"consecutive"} should match exactly one schema in oneOf.

    at validateRuleOptions (/project/node_modules/eslint/lib/config/config-validator.js:119:19)
    at Object.keys.forEach.id (/project/node_modules/eslint/lib/config/config-validator.js:162:9)
    at Array.forEach (<anonymous>)
    at validateRules (/project/node_modules/eslint/lib/config/config-validator.js:161:30)
    at Object.validate (/project/node_modules/eslint/lib/config/config-validator.js:239:5)
    at loadFromDisk (/project/node_modules/eslint/lib/config/config-file.js:516:19)
    at Object.load (/project/node_modules/eslint/lib/config/config-file.js:559:20)
    at Config.getLocalConfigHierarchy (/project/node_modules/eslint/lib/config.js:227:44)
    at Config.getConfigHierarchy (/project/node_modules/eslint/lib/config.js:179:43)
    at Config.getConfigVector (/project/node_modules/eslint/lib/config.js:286:21)
    at Config.getConfig (/project/node_modules/eslint/lib/config.js:329:29)
    at processText (/project/node_modules/eslint/lib/cli-engine.js:163:33)
    at CLIEngine.executeOnText (/project/node_modules/eslint/lib/cli-engine.js:620:17)
    at Transform.util.transform [as _transform] (/project/node_modules/gulp-eslint/index.js:50:20)
    at Transform._read (_stream_transform.js:185:10)
    at Transform._write (_stream_transform.js:173:12)
Emitted 'error' event at:
    at Transform.onerror (/project/node_modules/vinyl-fs/node_modules/readable-stream/lib/_stream_readable.js:558:12)
    at Transform.emit (events.js:180:13)
    at onwriteError (_stream_writable.js:425:12)
    at onwrite (_stream_writable.js:453:5)
    at Transform.afterTransform (_stream_transform.js:89:3)
    at Transform.util.transform [as _transform] (/project/node_modules/gulp-eslint/index.js:52:4)
    at Transform._read (_stream_transform.js:185:10)
    at Transform._write (_stream_transform.js:173:12)
    at doWrite (_stream_writable.js:410:12)
    at writeOrBuffer (_stream_writable.js:396:5)
    at Transform.Writable.write (_stream_writable.js:294:11)
    at write (/project/node_modules/vinyl-fs/node_modules/readable-stream/lib/_stream_readable.js:623:24)
    at flow (/project/node_modules/vinyl-fs/node_modules/readable-stream/lib/_stream_readable.js:632:7)
    at DestroyableTransform.pipeOnReadable (/project/node_modules/vinyl-fs/node_modules/readable-stream/lib/_stream_readable.js:664:5)
    at DestroyableTransform.emit (events.js:180:13)
    at emitReadable_ (/project/node_modules/vinyl-fs/node_modules/readable-stream/lib/_stream_readable.js:448:10)
```

**Comment by stevenvachon** [comment]
@Aladdin-ADD have you started working on this?

**Comment by Aladdin-ADD** [comment]
hi, @stevenvachon. sorry for the late!

This is working as designed: it has one option, which can be a string option or an object option. while it does not support mixed-option.

More details can be found [here](https://eslint.org/docs/rules/one-var#options).

> Object option:
> 
> "var": "always" requires one var declaration per function
> "var": "never" requires multiple var declarations per function
> "var": "consecutive" requires consecutive var declarations to be a single declaration
> ...

> Alternate object option:
> 
> "initialized": "always" requires one variable declaration for initialized variables per scope
> "initialized": "never" requires multiple variable declarations for initialized variables per scope
> "initialized": "consecutive" requires consecutive variable declarations for initialized variables to be a single declaration

**Comment by stevenvachon** [comment]
So it's impossible to configure ESLint the way I'd like. What does this mean for the future?

**Comment by Aladdin-ADD** [comment]
Please feel free to open an issue. it would really help us to have the information requested in this set of guidelines: [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) . Thanks!

**Comment by stevenvachon** [comment]
Okay, updated my original comment. Please update the tags.

**Comment by stevenvachon** [comment]
Updated my original comment and created #10234.

**Comment by stevenvachon** [comment]
@Aladdin-ADD please re-review as I've made changes.
